### PR TITLE
Expand WooPayments SSR entries to cover real HEs needs

### LINKED ISSUE #7122: SSR: Indicate what features/settings are enabled
### Description
Currently, the system status report information for WooPayments looks something like this:

```
### WooCommerce Payments ###

Version: 4.5.6
Connected to WPCOM: Yes
Blog ID: 123456789
Account ID: acct_a1b2c3d4e5f6g7h8
```

However, there are some issues that we encounter where it only affects merchants who have certain settings enabled (e.g., UPE, WooPay, Stripe Link, Multi-currency).

Some of these settings are not reported in our internal tools, or they are obfuscated to some degree in Looker or in the Audit Trail.

Listing what features/settings are enabled via the SSR will allow HEs to get a better determination of how WCPay is configured in wp-admin without having access to wp-admin.

Here are some options that might be particularly helpful for HEs:
- Enabled/disabled
- Test mode (enabled/disabled)
- UPE (enabled/disabled)
- List of APMs enabled (if UPE is enabled)
- Express Checkout methods (enabled/disabled) and what pages they are enabled on
- Auth + Capture (enabled/disabled)
- Fraud tools (basic/advanced)
- Multi-currency (enabled/disabled)
- Public key encryption (enabled/disabled)
- Logging (enabled/disabled)

### Acceptance criteria

* First requirement
* Second requirement
* Third requirement
* etc.

### Testing instructions

1. Go to '…'
2. Click on '…'
3. Scroll down to '…'
4. Observe expected behaviour

### PR DESCRIPTION
**Pull request by vladolaru**
Fixes #7122

### Changes proposed in this Pull Request

- Add all the new WooPayments data points mentioned in the issue (and some that were missed)
- Ensure **a sane conditional logic** between the WooPayments state and what data points are included in the SSR (especially when the payment gateway is not connected - no account data)
- Avoid SSR HEX (HE eXperience) quirks wherever possible (especially around comma-separated lists that on SSR export get separated by new lines - this is not appropriate for small items lists like the enabled payment methods list).
- Move the WooPayments SSR section at the top of the hooked sections (right after theme and overwritten templates).

Before:
Screenshot 2023-09-19 at 17 39 17

After with full details:
Screenshot 2023-09-19 at 17 41 05

After with no account connected:
Screenshot 2023-09-19 at 17 40 16

The exported SSR section looks like this:

```
### WooPayments ###

Version: 6.4.2
Connected to WPCOM: Yes
WPCOM Blog ID: 213164927
Account ID: acct_1NqYhAR4rnBgD8I0
Payment Gateway: Disabled
Test Mode: Enabled
UPE: Enabled (split)
UPE Enabled APMs: card,bancontact,giropay
WooPay: Enabled (checkout,cart)
WooPay Incompatible Extensions: No
Apple Pay / Google Pay: Enabled (product,cart)
Fraud Protection Level: advanced
Enabled Fraud Filters: AVS Verification,International IP Address,Order Items Threshold
Multi-currency: Enabled
Public Key Encryption: Disabled
Auth and Capture: Enabled
Documents: Enabled
Logging: Enabled
```

### Testing instructions

* Checkout the PR branch on your local client installation

#### Test with no WPCOM connection

* To simulate no Jetpack connection, temporarily add a `return false;` [here](https://github.com/Automattic/woocommerce-payments/blob/37572ec1222ff76dc4401d39d69a8e27faf0b991/includes/wc-payment-api/class-wc-payments-http.php#L130-L130).
* The WooPayments SSR section should look like this:
Screenshot 2023-09-19 at 18 09 41

#### Test with no account (WooPayments not connected to the WooPayments server)

* Remove the `return false;` added above
* From the WooPayments Dev Tools, enable the `Force the WCPay plugin to act as disconnected from the WCPay Server` option
* The WooPayments SSR section should look like this:
Screenshot 2023-09-19 at 18 12 05

#### Test with an account (WooPayments is connected to the WooPayments server)

* Uncheck the `Force the WCPay plugin to act as disconnected from the WCPay Server` from the WooPayments Dev Tools
* Make sure your local installation is connected by going through the WooPayments onboarding, if necessary
* The WooPayments SSR section should look like this:
Screenshot 2023-09-19 at 18 14 01

#### Smoke test

* Smoke test the WooPayments settings as much as you can (including enabling and disabling features via the WooPayments Dev Tools)
* The WooPayments SSR section should reflect the changes, and the information should make sense.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7236` or branch name `add/7122-more-details-to-woo-ssr` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/7122-more-details-to-woo-ssr) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 14cc4f719cb71e318e2ba079fde9f44abdbf81f4
- Build time: 2023-09-20 08:48:29 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by oaratovskyi — APPROVED** [review]
@vladolaru everything works as a charm! And looks amazing, so much helpful info!

I have only one question here - is it expected that Stripe Link is shown as one of the UPE enabled APMs? Or it's better to show it separately as in WCPay UI?
image

image

anyway - pre-approving!

File: includes/class-wc-payments-status.php [hunk 1]
```diff
@@ -67,39 +76,193 @@ public function debug_tools( $tools ) {
...
+					<tr>
+						<td data-export-label="WooPay Incompatible Extensions"><?php esc_html_e( 'WooPay Incompatible Extensions', 'woocommerce-payments' ); ?>:</td>
+						<td class="help"><?php echo wc_help_tip( esc_html__( 'Whether there are extensions active that are have known incompatibilities with the functioning of the new WooPay Express Checkout.', 'woocommerce-payments' ) ); /* phpcs:ignore WordPress.XSS.EscapeOutput.OutputNotEscaped */ ?></td>
+						<td><?php get_option( \WCPay\WooPay\WooPay_Scheduler::INVALID_EXTENSIONS_FOUND_OPTION_NAME, false ) ? esc_html_e( 'Yes', 'woocommerce-payments' ) : esc_html_e( 'No', 'woocommerce-payments' ); ?></td>
+					</tr>
+					<?php endif; ?>
+
+					<tr>
+						<td data-export-label="Apple Pay / Google Pay"><?php esc_html_e( 'Apple Pay / Google Pay Express Checkout', 'woocommerce-payments' ); ?>:</td>
+						<td class="help"><?php echo wc_help_tip( esc_html__( 'Whether the store has Payment Request enabled or not.', 'woocommerce-payments' ) ); /* phpcs:ignore WordPress.XSS.EscapeOutput.OutputNotEscaped */ ?></td>
```
Comment (oaratovskyi) [inline includes/class-wc-payments-status.php:179]: Is the Payment Request alternative name of Apple/Google Pay?

**Comment by oaratovskyi** [comment]
p.s. tested some fraud filters 🎉 
image

File: includes/class-wc-payments-status.php (hunk 1 above)
Comment (vladolaru) [reply includes/class-wc-payments-status.php:179]: It is the official name of the technology behind it. It is not limited to Apple Pay or Google Pay. The user browser decides.

**Comment by vladolaru** [comment]
Thank you, @oaratovskyi , for doing this review!

As for the Stripe Link question, I do not know, but internally we treat it as an APM, so I decided to leave it as such.

**Comment by frosso** [comment]
Thanks for working on this, @vladolaru ! This is great!

**Comment by vladolaru** [comment]
Thank you @frosso! Woo HEs need all the help they can get, especially these days.

**Review by fashuvo — COMMENTED** [review]
Thanks for working on this. I always wondered how the SSR info is generated and sent back for. After going through the changes in this PR and also exploring how the `render_status_report_section` is hooked into other places, now I have a better understanding of the process.

**Pull request merged**